grader/__init__.py

```diff
@@ -3,31 +3,35 @@
 grader: a Python module and command-line utility to grade applications.
 """
 import os
-import subprocess
+
+# current dir
+CWD = os.path.abspath(os.path.dirname(__file__))
 
 __name__ =        'grader'
 __description__ = 'a Python module and command-line utility to grade applications'
 __version__ =     '0.1'
 __url__ =         'https://github.com/ASPP/grader'
 __author__ =      'grader contributor'
 __license__ =     'GPLv3+'
-__revision__ =    'N/A'
```

Comment:
Hmm, but then __revision__ is not set anymore?

Reply:
`__revision__` is not used in `setup.py` so it's not a necessary value

if needed a user can call the new `revision()` function and get that number. That avoids launching a subprocess that tries to call `git` every time you import `grader`

Reply:
Then let's just drop this. It's not like grader has any external users...

Reply:
just `__revision__` or the `revision()` function as well?

Reply:
Both, I think. It looks like `__revision__` was added in 335682b46f2ab8bece1ca9c4001a5dfeece8d053 but never used (see `metadata_vars` in that commit).

Reply:
I'd like to have an handle on the git commit the version of grader I am using comes from. As simple as that. That is especially useful if we start tagging grader.conf and store them for later analysis. I don't like the `revision` function, or better, I'd like the functiuon to be run once at `setup.py`, when the grader wheel is created, and then the output of this gets hard-coded in __revision__. This is what `numpy` and `scipy` are doing.

Reply:
To be clear: I don't envision the `__revision__` attribute as something used on a daily basis by a "user" of grader. I expect it to be part of richer metadata for provenance tracking. I agree that this is mostly overkill for grader, but why not try to have the "right" solution? It's not like it requires a lot of over-engineering…

Reply:
I agree with @otizonaizit, this is something I always have wanted to know eg. with scikit-image. I'll definitely port this to that project!